qiskit/qpy/type_keys.py

```diff
@@ -524,3 +524,21 @@ def assign(cls, obj):
     @classmethod
     def retrieve(cls, type_key):
         raise NotImplementedError
+
+
+class Encoding(TypeKeyBase):
```

Comment:
I'm wondering if we should name this `SymExprEncoding` or something a bit more descriptive. I'm just worried the name `Encoding` is too generic because if I saw that without the context of this PR I'd think it was like `UTF-16`.

Reply:
Being more specific with the class name would surely not hurt (especially if we wanted to add another encoding type in the future), so changed in [b479cf0](https://github.com/Qiskit/qiskit/pull/10820/commits/b479cf061fa95de432e7d0c76e41b20832ab5222).